src/os/freertos/include/AbstractTask.tpp

```diff
@@ -19,9 +21,8 @@ AbstractTask<stackSize>::~AbstractTask() {
 template <unsigned int stackSize>
 bool AbstractTask<stackSize>::start() {
     if (!m_taskRunning) {
-        m_taskHandle = xTaskCreateStatic(wrapper, m_taskName, stackSize, this, m_priority,
-                                         m_stackArray.data(), &m_taskBuffer);
         m_taskRunning = true;
+        vTaskResume(m_taskHandle);
```

Comment:
Hmm I see what you are trying to do... I'd add a check on the taskHandle, if it is null, create the task, if it isn't and the m_running flag is false, resume it, else do nothing.

Also do we really want to return false if the task was already running?

Reply:
Since we create from static, we know the task handle is not null, though I could add a check but how do I handle it if it's NULL? Should I assert?

Reply:
For the false, that's the interface that we have as of now, true if the operation was successfull,  false if not (i.e. already started)